File: 2019 maj/python/1.py
with open("liczby.txt", "r") as f:
    liczby = list(map(int, f.read().strip().split('\n')))

s = set()
i = 1
while i <= 100000:
    s.add(i)
    i *= 3

ans = 0
for i in liczby:
    if i in s:
        ans += 1
print(ans)

File: 2019 maj/c++/1.cpp
#include <iostream>
#include <fstream>
#include <set>
using namespace std;

set <int> s;

int main()
{
    ifstream in("liczby.txt");
    
    for (int i = 1; i <= 100000; i *= 3)
        s.insert(i);
    
    int ans = 0;
    for (int i = 1; i <= 500; i++)
    {
        int a;
        in >> a;
        if (s.count(a))
            ans++;
    }
    cout << ans << '\n';
}